### DISCUSSION TITLE
Stock items label templates

### DISCUSSION THREAD

**Discussion by anonkwe** [inventree/InvenTree#5997] [Q&A] [answered]
Hi, is there a way to incorporate purchase order fields in stock items label templates? it would be very useful for us to display stock's PO reference number. i guess the field would look like the one used in the po template ? (   {{ reference }}   ) .  is there any kind of lexical or repertory where usable tags.. or fields (not sure how to call those) are listed?

**Comment by miggland** [comment, marked as answer]
PO Reference is in the item:
`{{ item.purchase_order.reference }}`

**Reply by anonkwe** [reply]
@miggland  thanks a lot ! is there any kind of list with the possible labels somewhere out there?

**Reply by SchrodingersGat** [reply]
@anonkwe there is a [list on the documentation site](https://docs.inventree.org/en/stable/report/labels/stock_labels/#context-data) but it is not comprehensive. We would happily accept a PR to improve the documentation around the available context variables

**Reply by wolflu05** [reply]
You can also use a devcontainer with the debugger to pause in a label template. That way you can inspect all available variables.

**Reply by miggland** [reply]
There is also a debug extension for Jinja which shows all available context and filters. However, it's not activated (I think..).

https://jinja.palletsprojects.com/en/3.1.x/extensions/#debug-extension
